src/main/java/org/launhcode/healthynutrition/controllers/ForgotPasswordController.java

```diff
@@ -0,0 +1,145 @@
+package org.launhcode.healthynutrition.controllers;
+
+import net.bytebuddy.utility.RandomString;
+import org.launhcode.healthynutrition.data.UserRepository;
+import org.launhcode.healthynutrition.models.User;
+import org.launhcode.healthynutrition.models.service.CustomerService;
+import org.launhcode.healthynutrition.models.service.UserNotFoundException;
+import org.launhcode.healthynutrition.models.service.Utility;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.repository.query.Param;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import javax.servlet.http.HttpServletRequest;
+import java.io.UnsupportedEncodingException;
+
+@Controller
+public class ForgotPasswordController {
+    @Autowired
+    private JavaMailSender mailSender;
+
+    @Autowired
+    private CustomerService customerService;
+
+    @Autowired
+    private UserRepository userRepo;
+
+    @ModelAttribute
+    @GetMapping("/forgot_password")
+    public String showForgotPasswordForm() {
+        return "forgot_password";
+    }
+
+    @PostMapping("/forgot_password")
+    public String processForgotPassword(Model model, HttpServletRequest request) {
+        String email = request.getParameter("email");
+        String token = RandomString.make(30);
+
+        try {
+            customerService.updateResetPasswordToken(token, email);
+            String resetPasswordLink = Utility.getSiteURL(request) + "/reset_password?token=" + token;
+            sendEmail(email, resetPasswordLink);
+            model.addAttribute("message", "We have sent a reset password link to your email. Please check.");
+
+
+        } catch (UserNotFoundException ex) {
+            model.addAttribute("error", ex.getMessage());
+        } catch (UnsupportedEncodingException | MessagingException e) {
+            model.addAttribute("error", "Error while sending email");
+        }
+
+        return "forgot_password";
+    }
+
+    public void sendEmail(String recipientEmail, String link)
+            throws MessagingException, UnsupportedEncodingException {
+        MimeMessage message = mailSender.createMimeMessage();
+        MimeMessageHelper helper = new MimeMessageHelper(message);
+
+        helper.setFrom("contact@shopme.com", "Shopme Support");
```

Comment:
shopme support does not seem related to this app :p

Reply:
from this site only I'm getting me reset password link..
![image](https://user-images.githubusercontent.com/72172122/115443077-6c7be980-a1d8-11eb-93c4-a0044545ef60.png)

Reply:
Yeah, I'm saying you should change the name to something related to this app instead of copying/pasting from the tutorial